src/OpenTelemetry.Api/Internal/NullableAttributes.cs

```diff
@@ -0,0 +1,37 @@
+// <copyright file="NullableAttributes.cs" company="OpenTelemetry Authors">
```

Comment:
Please check https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry/Internal/Shims/NullableAttributes.cs

I think that content from this file should be merged with existing one.

Reply:
These should absolutely be merged!

I didn't know this file was here! This is probably what was causing conflicts in my nullable branch. Thanks for pointing this out! :)

Reply:
I've run into an issue merging these two files and I'm still exploring options before I make a recommendation.

Both `OpenTelemetry` and `OpenTelemetry.Api` projects need access to nullables.
OTel has a build dependency on OTel.Api.
OTel.Api makes internals visible to OTel.

The blocker is that OTel targets `net6.0;netstandard2.1;netstandard2.0;net462` where as OTel.Api targets `netstandard2.0;net462`.
OTel (net6 and netstardard2.1) will take a dependency on OTel.Api (netstandard2.0).
And then OTel (net6 and netstardard2.1) has a conflict with runtime. ☹️☹️☹️

> Error	CS0433	The type 'AllowNullAttribute' exists in both 'OpenTelemetry.Api, Version=1.0.0.0, Culture=neutral, PublicKeyToken=7bd6737fe5b67e3c' and 'System.Runtime, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'	OpenTelemetry (net6.0)

Reply:
The fix is to add netstandard2.1 to the OTel.Api project. I'll bring this up in the SIG meeting tomorrow.

Reply:
Discussed this in today's SIG and there are no blockers with this approach. :)


It would be preferred if we can move shared files to a separate project, but this only works if we can also remove the internals visible to between OTel and OTel.Api. I'm going to take a day to explore if this would work.